src/expr/sygus_grammar.h

```diff
@@ -120,15 +133,14 @@ class SygusGrammar
   Node purifySygusGNode(const Node& n,
                         std::vector<Node>& args,
                         std::vector<TypeNode>& cargs) const;
-
+  /** The rules */
+  std::map<Node, std::vector<Node>> d_rules;
```

Comment:
```suggestion
  std::unordered_map<Node, std::vector<Node>> d_rules;
```
I just use `d_ntSyms` if I care about order...